src/ast/passes/parser.h

```diff
@@ -52,8 +53,13 @@ inline std::vector<Pass> AllParsePasses(
   passes.emplace_back(CreateApExpansionPass());
   passes.emplace_back(CreateParseTracepointFormatPass());
   passes.emplace_back(CreateBuiltinsPass());
+  passes.emplace_back(CreateProbeExpansionPass());
   passes.emplace_back(CreateFieldAnalyserPass());
   passes.emplace_back(CreateClangParsePass(std::move(extra_flags)));
+  // N.B. We need to run the clang parse pass before attempting to resolve
+  // the `args` for tracepoints and expand probes but we need to resolve
+  // the `args` for other probe types before the clang parse pass
```

Comment:
This is quite unfortunate but correct. Let's see if the new expansion approach where we expand everything and dedup later will allow us to resolve this somehow (I should really bite the bullet and start working on it finally).